go/batch-submitter/service.go

```diff
@@ -99,6 +105,15 @@ func (s *Service) eventLoop() {
 
 	name := s.cfg.Driver.Name()
 
+	if s.cfg.ClearPendingTx {
+		for {
```

Comment:
I suggest we limit this loop to a fixed number of iterations, and `log.Fatalf` if we don't successfully clear pending txs. I say this because it'll be noisier in our monitoring, and force us to take a look rather than spinning forever if there's ever a permanent failure.

Reply:
fixed 👍